Week 3 Pull Request from dev into main

Antonov333 opened the pull request.

Antonov333 committed:
Merge pull request #24 from Antonov333/dev

Update developer's branch with actual Dev branch

Antonov333 committed:
Created findByUserName method in UserRepository

Antonov333 committed:
Merge pull request #25 from Antonov333/2-serhiy

findByUserName method created in UserRepository

Antonov333 committed:
Cleaned AuthController from commented code

Antonov333 committed:
UserRepository fixed: findByUserName method renamed to findByEmail

Antonov333 committed:
Merge pull request #26 from Antonov333/2-serhiy

UserRepository fixed

Antonov333 committed:
Created JavaKidsUserDetails and JavaKidsUserDetailsManager classes, updated AuthServiceImpl and WebSecurityConfig. Next step: complete realization of user entity creation logic

Antonov333 committed:
JavaKidsUserDetailsManager: userRepository field fixed (made final)

Antonov333 committed:
javadoc and some code added

vpold19 committed:
experiment with security 2.0

Antonov333 committed:
AuthServiceImpl: completed register method
User: javadoc added

Antonov333 committed:
AuthServiceImpl: User class corrected to ours in register method
User: javadoc added

Antonov333 committed:
User: javadoc mistyping corrected
JavaKidsUserDetailsManager: declared createUser(RegisterDto registerDto) <-- NO OVERRIDE parent interface method (polymorphism)

Antonov333 committed:
User: javadoc mistyping corrected
JavaKidsUserDetailsManager: declared createUser(RegisterDto registerDto) <-- NO OVERRIDE parent interface method (polymorphism)

Antonov333 committed:
Merge remote-tracking branch 'origin/2-w3-serhiy' into 2-w3-serhiy

Antonov333 committed:
JavaKidsUserDetailsManager: completed createUser(RegisterDto registerDto) (NOT OVERRIDING parent interface method) and used it with AuthServiceImpl

Antonov333 committed:
AuthServiceImpl: improved code, provided password encoding

Antonov333 committed:
SecurityFilterChain changed a little

Antonov333 committed:
Making methods to make Advertisements controller work

Antonov333 committed:
Composed Advertisements controller and AdvertisementsService, created some utility classes and updated AdRepository. Need to complete updateAd method in AdvertisementsService

Antonov333 committed:
Current commit

Antonov333 committed:
Current commit

Antonov333 committed:
Learning apply @PreAuthorize annotation with Advertisements controller and AdvertisementsService

Antonov333 committed:
Implementing biz logic to CommentsController, so created CommentsService, wrote method to CommentRepository, created util class CommentsDtoFound, added javadoc to some classes

Antonov333 committed:
Business logic implemented to CommentsController and CommentsService

Antonov333 committed:
Merge pull request #27 from Antonov333/2-w3-serhiy

Graduate Work Week 3

Antonov333 committed:
Security features added to Advertisements controller and AdvertisementsService, unused controller RegistrationController deleted

Antonov333 committed:
Merge pull request #28 from Antonov333/2-w3-serhiy

Security features added to Advertisements controller and Advertisemen…

Antonov333 committed:
User registration and login fixed

Antonov333 committed:
Merge pull request #30 from Antonov333/2-w3-serhiy

User registration and login fixed

Antonov333 committed:
Improving code...

Antonov333 committed:
User to UserDto mapper fixed, AdMapper class changed a little

Antonov333 committed:
Endpoint POST /ads still needs fix...

Antonov333 committed:
Endpoint POST /ads fixed with image field keeping path to picture of item

Antonov333 committed:
Merge pull request #31 from Antonov333/2-w3-serhiy

Follwong up Week 3 Coach's consideration...

Antonov333 committed:
Midway commit...

Antonov333 committed:
Fixed endpoints: POST /ads, DELETE /ads/{id}

Antonov333 committed:
Tonight commit

Antonov333 committed:
Midday commit...

Antonov333 committed:
Changed arg names in Registration controller and AuthServiceImpl service for better code understanding

Antonov333 committed:
Fixed time of comment creation: apply right type of created_at field of Comment entity and implement correct right

Antonov333 committed:
Fixed POST /users/set_password endpoint: setPassword method provided with necessary logic

Antonov333 committed:
Merge pull request #32 from Antonov333/2-w3-serhiy

Week 3, 4: improved changing password, comment creation time and some other features

Antonov333 committed:
Working with pictures...

Antonov333 committed:
Merge pull request #33 from Antonov333/2-w3-serhiy

Working with pictures...

Antonov333 committed:
Composed ImagesController with endpoint to deliver images to frontend application, completed setAvatar method in UserServiceImpl service class

Antonov333 committed:
Merge pull request #34 from Antonov333/2-w3-serhiy

Composed ImagesController with endpoint to deliver images to frontend…

Antonov333 committed:
Merge branch 'main' into dev

Antonov333 committed:
Changed methods changeAvatar in AvatarController and setAvatar in UserServiceImpl

Antonov333 committed:
Fixed showing avatars with UI

Antonov333 committed:
Merge pull request #35 from Antonov333/2-w3-serhiy

Fixed avatar image storing in database and delivery to frontend

Antonov333 committed:
Fixed saving comments in database and delivery to frontend

Antonov333 committed:
Merge pull request #36 from Antonov333/2-w3-serhiy

Fixed saving comments in database and delivery to frontend

Antonov333 committed:
Made sure item pictures delivered to frontend

Antonov333 committed:
Merge pull request #37 from Antonov333/2-w3-serhiy

Made sure item pictures delivered to frontend

artem-petrosyan commented on src/main/java/ru/skypro/homework/model/Ad.java:
```diff
@@ -1,13 +1,15 @@
 @Table(name = "ads")
 @Data
+@NoArgsConstructor
 public class Ad {
```
Добавьте вот такое поле:
```
    @ManyToOne
    @JoinColumn(name = "Author", updatable = false, insertable = false)
    private User authorEntity;
```

Antonov333 replied:
Добавили

artem-petrosyan commented on src/main/java/ru/skypro/homework/mapping/AdMapper.java:
```diff
@@ -3,10 +3,37 @@
+    Ad adDtoToad(AdDto dto);
+
+    Ad CrOUpdToAd(CreateOrUpdateAdDto createOrUpdateAdDto);
+
```
```
    @Mapping(source = "authorEntity.name",target = "authorFirstName")
    @Mapping(source = "authorEntity.surname",target = "authorLastName")
    @Mapping(source = "authorEntity.phoneNumber",target = "phone")
    @Mapping(source = "authorEntity.email",target = "email")
    ExtendedAdDto adToExtendedAdDto (Ad ad);
```

И такой маппер

Antonov333 replied:
Добавили

artem-petrosyan commented on src/main/java/ru/skypro/homework/controller/Advertisements.java:
```diff
@@ -1,61 +1,91 @@
+     * @return DTO of advertisements with given id
      */
     @GetMapping("/ads/{id}")
     public ResponseEntity<AdDto> getAds(@PathVariable long id) {
-        return new ResponseEntity<>(new AdDto(), HttpStatus.OK);
+        AdFound adFound = advertisementsService.getAdById(id);
+        return new ResponseEntity<>(AdMapper.INSTANCE.adToDto(adFound.getAd()), adFound.getHttpStatus());
     }
```
```
    @GetMapping("/ads/{id}")
    public ResponseEntity<ExtendedAdDto> getAds(@PathVariable long id) {
        AdFound adFound = advertisementsService.getAdById(id);
        return new ResponseEntity<>(AdMapper.INSTANCE.adToExtendedAdDto(adFound.getAd()), adFound.getHttpStatus());
    }

```
Тут используем ExtendedAdDto

Antonov333 replied:
Исправили возращаемый тип

artem-petrosyan commented on src/main/java/ru/skypro/homework/controller/Advertisements.java:
```diff
@@ -66,70 +96,114 @@ public ResponseEntity<AdDto> getAds(@PathVariable long id) {
-                                 @PathVariable long id) {
-        return new ResponseEntity<>(new AdDto(), HttpStatus.OK);
+                                    @PathVariable long id, @RequestBody CreateOrUpdateAdDto updatedAdContent,
+                                    String userName) {
+
+        return advertisementsService.updateAd(id, updatedAdContent, userName);
 
     }
```
```suggestion
                                    Authentication authentication) {

        return advertisementsService.updateAd(id, updatedAdContent, authentication.getName());

    }
```

Antonov333 replied:
Исправили

artem-petrosyan commented on src/main/java/ru/skypro/homework/controller/Advertisements.java:
```diff
@@ -66,70 +96,114 @@ public ResponseEntity<AdDto> getAds(@PathVariable long id) {
+                                                     @PathVariable("userId") Integer userId) {
+        Images image = Images.builder()
+                .pictureType(PictureType.USER_AVATAR)
+                .userId(userId)
```
```suggestion
                .userId(userId.longValue())
```

Antonov333 replied:
Исправили

artem-petrosyan commented on src/main/java/ru/skypro/homework/service/AdvertisementsService.java:
```diff
@@ -0,0 +1,288 @@
+        Images itemPicture = Images.builder().
+                pictureType(PictureType.ITEM_PICTURE).
+                adId(newAd.getPk()).
+                userId(newAd.getAuthor()).
```
```suggestion
                userId(newAd.getAuthor().longValue()).
```

Antonov333 replied:
Добавили

artem-petrosyan commented on src/main/java/ru/skypro/homework/service/AdvertisementsService.java:
```diff
@@ -0,0 +1,288 @@
+
+        itemPicture = imagesRepository.save(itemPicture);
+
+        newAd.setImage("/image/" + itemPicture.getId()); // provide image field with the path to endpoint delivering picture
```
```suggestion
        newAd.setImage("/images/" + itemPicture.getId()); // provide image field with the path to endpoint delivering picture
```
url неправильный

Antonov333 replied:
URL исправили, изменения загружены в dev

artem-petrosyan commented on src/main/java/ru/skypro/homework/model/Images.java:
```diff
@@ -1,12 +1,23 @@
 @Table(name = "images")
+@NoArgsConstructor
+@AllArgsConstructor
 public class Images {
```
В этом классе замените все `long` на обёртки `Long`

Antonov333 replied:
Заменили

artem-petrosyan commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -1,21 +1,28 @@
 @CrossOrigin(value = "http://localhost:3000")
 @RestController
 @RequestMapping("/users")
 public class UserController {
```
Не хватает методов:
PATCH /users/me
PATCH /users/me/image

Antonov333 replied:
Добавляем!

Antonov333 replied:
Методы добавлены

artem-petrosyan commented on src/main/java/ru/skypro/homework/service/impl/UserServiceImpl.java:
```diff
@@ -41,5 +78,43 @@ public AvatarDto setAvatar(MultipartFile avatar) throws IOException {
+        }
+        User user = userRepository.findByEmail(principal.getName()).get();
+        Images avatarPicture = Images.builder()
+                .userId(user.getId())
```
```suggestion
                .userId(user.getId().longValue())
```

Antonov333 replied:
Добавили

Antonov333 committed:
Fixed saving item pictures to database and retrieving to frontend

Antonov333 committed:
Merge pull request #38 from Antonov333/2-w3-serhiy

Fixed saving item pictures to database and retrieving to frontend

Antonov333 committed:
Fixed types of variables and values

Antonov333 committed:
Fixed endpoints:
PATCH /users/me
PATCH /users/me/image

Antonov333 committed:
Merge pull request #39 from Antonov333/2-w3-serhiy

Followed up Coach's review

Antonov333 committed:
Fixed:
Ad and User entities relationship
Composed mapper from Ad to ExtendedAdDto
Return type of getAds(@PathVariable long id)

Antonov333 committed:
Fixed updateAds method in Advertisements controller

Antonov333 committed:
Merge pull request #40 from Antonov333/2-w3-serhiy

Fixed entity and methods in controller and service for advertisements, added mapper
